### PR TITLE
Timeouts on DB futures, classpath fixes, gradle lint

### PR DESCRIPTION
**Pull request by tomnis**
- a few minor things i discovered while investigating another issue
- DB-related Futures (queries and opening connection) have a 90 second timeout. is it worth making a config property for this?
- exclude logback from our dependencies (gatling). this fixes a warning about multiple slf4j concrete bindings.
- other dependency cleanup, i removed unused dependencies (with the help of nebula gradle-lint), and declare explicit dependencies on transitive libraries we are using directly (eg scala-reflect and typesafe config)
- changed h2 and mysql-connector-java to runtime dependencies (the latter was suggested by gradle-lint)
- since our poms will look fairly different now, we should bump major version for the next release

### CODE REVIEW

File: src/main/scala/com/workday/warp/persistence/Tables.scala [hunk 1]
```diff
@@ -18,6 +18,8 @@ object TablesLike extends model.TablesLike
 object Drivers {
 
   val mysql: String = "com.mysql.jdbc.Driver"
+  // see https://dev.mysql.com/doc/connector-j/8.0/en/connector-j-api-changes.html
```
Comment (rooeque) [inline src/main/scala/com/workday/warp/persistence/Tables.scala]: 👍

File: versionInfo.gradle [hunk 2]
```diff
@@ -37,47 +37,34 @@ def gatlingVersion() {
...
     , h2: '1.4.191'
-    , hamcrest: '1.3'
     , influxDbClient: '2.14'
-    , jacksonDataBind: '2.7.4'
-    , jbake: '1.0.0'
-    , json4s: '3.4.2'
     , junit: '4.12'
     , math3: '3.5'
-    , mysqlConnector: '5.1.46'
+    , mysqlConnector: '5.1.47'
```
Comment (rooeque) [inline versionInfo.gradle]: is there anything preventing us from using mysqlJ 8.0?

Comment (tomnis) [reply versionInfo.gradle]: it didnt work haha. broke the code generator, i dug into it a bit, and somehow slick `MySQLProfile.defaultTables` returns an empty collection.

Comment (rooeque) [reply versionInfo.gradle]: oh right lol

**Pull request merged**